Source/UZKArchive.m

```diff
@@ -880,16 +884,8 @@ - (BOOL)extractBufferedDataFromFile:(NSString *)filePath
                 NSMutableData *data = [NSMutableData dataWithLength:bufferSize];
                 int bytesRead = unzReadCurrentFile(welf.unzFile, data.mutableBytes, (unsigned)bufferSize);
                 
-                if (bytesRead < 0) {
-                    NSString *detail = [NSString localizedStringWithFormat:NSLocalizedStringFromTableInBundle(@"Failed to read file %@ in zip", @"UnzipKit", _resources, @"Detailed error string"),
-                                        info.filename];
-                    UZKLogError("Error reading data (code %d): %{public}@", bytesRead, detail);
-                    [welf assignError:&strongInnerError code:bytesRead
-                               detail:detail];
-                    break;
-                }
-                else if (bytesRead == 0) {
-                    UZKLogDebug("Done reading file");
+                if (bytesRead <= 0) {
```

Comment:
@amosavian What prompted you to check <= 0 instead of < 0? I'm treating that case as the file read being done. If it's 0, doesn't that mean there was no error, since the negative numbers are error codes? Let's definitely go with my code inside the if block, but I'd like to understand why you changed the condition.

Reply:
Because we must break loop either if no more bytes is available or an error is occured. I moved error creation logic out of the loop.

Reply:
Gotcha